In [2]:
import numpy as np
from scipy.stats import ttest_1samp

We have 100 patients with average value of sugar in blood being 6.06 with std=1.43. 
According to standards, the normal value of sugar is 5.0. Test whether the average level of those patients is normal

H0: avg(patients)=5.0

H1: it is not

In [28]:
patients=np.random.normal(5.6,1.6,1000000000)

In [29]:
patients.mean()

5.600045645081497

In [25]:
patients.std()

1.60001640888848

In [6]:
ttest_1samp(patients,5.0)

Ttest_1sampResult(statistic=7.304492048734628, pvalue=7.172430704204346e-11)

In [11]:
from scipy.stats import t

In [15]:
t.interval(0.95,df=99)

(-1.9842169515086827, 1.9842169515086827)

Based on our statistics value, the absolute value of our statistics is higher than critical value (1.98), therefore we reject Null Hypothesis, therefore the average concentration of sugar is not normal.

Similarly we can take a look on pvalue (0.0). This value is smaller than 0.05 (1-confidence level). Therefore there is almost no chance that we are wrong by rejecting null hypothesis. So, we reject it and the value of 6 is not normal

Lets calculate the t statistics manually

In [ ]:
t=(x-m)/std/sqrt(n)

In [18]:
t_data=(6.06-5)/(1.43/10)
t_table=t.interval(0.95,df=99)[1]

In [19]:
if abs(t_data)>t_table:
    print('Reject Null Hypothesis')
else:
    print('You cant reject H0')

Reject Null Hypothesis


I have a sample of 100 people in Montreal. THe mean is 5.6 std 1.6. At the same time I know that among all Canadians the average is 5.0. Is this sample representative?

## One-sided test

We have 100 patients with average value of sugar in blood being 5.66 with std=1.44. 
According to standards, the normal value of sugar is 5.0. Test whether the average level of those patients is higher than normal with 5% of significance level

H0: avg(patients)=5.0

H1: it is higher

In [6]:
patients=np.random.normal(5.6,1.6,100)

In [7]:
patients.mean()

5.660887554218029

In [8]:
patients.std()

1.4393680010008012

In [9]:
ttest_1samp(patients,5.0)

Ttest_1sampResult(statistic=4.568496821729001, pvalue=1.4181290356689762e-05)

In [13]:
t.interval(0.90,df=99)

(-1.6603911559963902, 1.6603911559963895)

I compare 4.57 and 1.66. Since 4.57>1.66, I can reject Null Hypothesis

At the same time I can just take a look on my pvalue. It is smaller than 10%. Thats enough

## One-sided test

We have 100 patients with average value of sugar in blood being 5.66 with std=1.44. 
According to standards, the normal value of sugar is 5.0. Test whether the average level of those patients is lower than normal with 5% of significance level

H0: avg(patients)=5.0

H1: it is lower

In [14]:
ttest_1samp(patients,5.0)

Ttest_1sampResult(statistic=4.568496821729001, pvalue=1.4181290356689762e-05)

In [15]:
t.interval(0.90,df=99)

(-1.6603911559963902, 1.6603911559963895)

I compare 4.57 and -1.66. SInce 4.57>-1.66, I cant reject Null Hypothesis

# 2 sample tests
## Paired

In [16]:
import pandas as pd

In [17]:
data=pd.read_csv('https://s3-eu-west-1.amazonaws.com/ih-materials/uploads/data-static/data/module-2/blood_pressure.csv')

In [19]:
data.mean()

before    136.536953
after      99.236700
dtype: float64

In [21]:
data['difference']=data.eval('after-before')

H0: before=after => difference=0

H1 : difference!=0

In [24]:
ttest_1samp(data.difference,0)

Ttest_1sampResult(statistic=-27.291841767560236, pvalue=7.303035069608042e-48)

In [25]:
from scipy.stats import ttest_rel

In [26]:
ttest_rel(data.after,data.before)

Ttest_relResult(statistic=-27.291841767560236, pvalue=7.303035069608042e-48)

In [27]:
data=pd.read_csv('https://s3-eu-west-1.amazonaws.com/ih-materials/uploads/data-static/data/module-2/ab_test.csv')

In [28]:
data

a      b
0    0.27  13.61
1    6.08  21.53
2   13.74   9.23
3    9.70   5.36
4    7.00  12.90
5    4.77  10.83
6    2.56  10.70
7    7.24  15.57
8    7.30  10.42
9    0.86  18.90
10   3.10  23.95
11  10.31   5.70
12  17.42  12.65
13  14.03   9.78
14   9.09  12.83
15  13.29  18.61
16  16.55  12.94
17   8.14   5.85
18  13.99  22.92
19  10.99  21.46
20  18.35   3.11
21   3.34  15.52
22  11.27   1.51
23  16.02  24.00
24   9.47  14.07
25  11.62  11.97
26   6.83   2.31
27  11.57   2.66
28  20.90  14.14
29   1.97  22.63
30  15.67  10.95
31  19.08   0.59
32   4.60  15.14
33   8.21  21.97
34   6.29  20.22
35  16.26  11.13
36  13.51  13.74
37   8.63   7.02
38   2.73   9.94
39   9.07   7.99
40   6.56  20.37
41  13.31   6.46
42   4.08  11.66
43   9.07  18.62
44   6.65  19.49
45   3.73   2.02
46   1.01   8.48
47  14.57  12.62
48   5.20  13.22
49  11.77   4.01

H0: there is no difference. a=b

H1: a!=b

In [29]:
data.isna().sum()

a    0
b    0
dtype: int64

In [30]:
data.mean()

a     9.3554
b    12.4660
dtype: float64

In [31]:
data.std()

a    5.235270
b    6.491255
dtype: float64

Same sample size, kinda equal std
1. variance are the same
2. variance are kinda same (0.5<s1/s2<2)
3. variances are different

In [32]:
from scipy.stats import ttest_ind

In [35]:
ttest_ind(data.a,data.b, equal_var=True)

Ttest_indResult(statistic=-2.637533181209767, pvalue=0.009713140852447347)

In [36]:
ttest_ind(data.a,data.b, equal_var=False)

Ttest_indResult(statistic=-2.637533181209767, pvalue=0.009776243024828825)